### PR TITLE
Refactor pages - layout and navigation

### PR DESCRIPTION
**Pull request by gsmendoza**
## Blockers 

Depends on https://github.com/nebulab/solidus_starter_frontend/pull/201 to be merged first.

## Goal

As a solidus starter frontend user
I want the pages to be refactored 
So that they would be easier to extend and update

## Implementation

I drilled down the implementation of the PR from the highest level templates (`spree_application.html.erb`, `products/index`, `taxons/show`) down to individual components.

## How Has This Been Tested?

Manually tested. HTML diffs recorded in git logs.

### http://locahost:3000

image

### http://localhost:3000/products

image

### http://localhost:3000/t/categories/clothing

image

## Types of changes

- N/A: Bug fix (non-breaking change which fixes an issue)
- N/A: New feature (non-breaking change which adds functionality)
- N/A: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- N/A: My change requires a change to the documentation.
- N/A: I have updated the documentation accordingly.

### CODE REVIEW

File: templates/app/views/spree/shared/locale/_locale_selector.html.erb [hunk 1]
```diff
@@ -12,14 +12,10 @@
...
-      <%= render(
-        "spree/components/forms/inputs/select_input",
-        name: :switch_to_locale,
-        options: available_locales,
-        selected: I18n.locale,
-        label: I18n.t('spree.i18n.language'),
-        classes: ['locale-selector__select-input']
-      ) %>
+      <div class="locale-selector__select-input select-input">
+        <%= label_tag :switch_to_locale, "#{t('spree.i18n.language')}:" %>
```
Comment (gsmendoza) [inline templates/app/views/spree/shared/locale/_locale_selector.html.erb:16]: I updated the label tag to point to the `:switch_to_locale` select tag.

File: templates/app/views/spree/layouts/_header.html.erb [hunk 2]
```diff
@@ -3,6 +3,8 @@
     <%= render 'spree/layouts/top_bar', taxons: taxons, taxon: taxon %>
   </div>
   <div class="header__primary-navigation">
-    <%= render 'spree/shared/navigation/category_navigation' %>
+    <div class="category-navigation">
+      <%= render partial: 'spree/components/navigation/categories' %>
```
Comment (kennyadsl) [inline templates/app/views/spree/layouts/_header.html.erb]: We don't need `partial:` here, right?

**Review by kennyadsl — APPROVED** [review]
Left a non-blocking comment, nice cleanup

File: templates/app/views/spree/layouts/_header.html.erb (hunk 2 above)
Comment (gsmendoza) [reply templates/app/views/spree/layouts/_header.html.erb]: This was removed in a future commit :)

**Pull request merged**